spec/requests/categories_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+describe "Categories pages" do
+  let(:user) { FactoryGirl.create(:user) }
+  let!(:category) { FactoryGirl.create(:category, user: user) }
+
+  describe 'authorized user' do
+    before { sign_in(user) }
+
+    it do
+      visit '/categories'
+      expect(page).to have_content('New category')
+    end  
+
+    it do
+      visit '/categories'
+      expect(page).to have_content('ALL CATEGORIES LIST')
+    end 
+
+    it do
+      visit '/categories'
+      expect(page).to have_content('Edit')
+      expect(page).to have_content('Destroy')      
+      expect(page).to have_content('Show')
+    end 
+  end
+
+  describe 'unauthorized user' do
+    it "redirects to home page" do
+      visit '/categories'
```

Comment:
move `visit '/categories'` to `before` block and delete from tests. it will help get rid of duplications